pkg/kapp/logs/pod_container_log.go

```diff
@@ -140,6 +163,16 @@ func (l PodContainerLog) readyToGetLogs() bool {
 	if err != nil {
 		return false
 	}
+	containerStatuses := pod.Status.ContainerStatuses
+	for _, containerStatus := range containerStatuses {
+		//We should focus only on the status of this particular container.
+		if l.container != containerStatus.Name {
+			continue
+		}
+		if containerStatus.State.Running != nil {
```

Comment:
Please provide your comments on this condition `containerStatus.State.Running != nil`. We need to make a decision on when to start streaming the logs. If we dont check for any condition, then it stream the older logs until the container is up leading to bad user experience and repeating the last few lines of previous container.

Reply:
I am not fully aware how and when this gets populated, but based on my understanding so far, this seems fine to me.